crates/telio-lana/src/event_log_moose.rs

```diff
@@ -98,109 +99,167 @@ pub fn init_context_info() {
+                    tracker_info.user.subscription.current_state.merchant_id =
                         get_i32_field(foreign_state, "merchant_id");
-                    user_info.subscription.current_state.payment_amount =
+                    tracker_info.user.subscription.current_state.payment_amount =
                         get_f32_field(foreign_state, "payment_amount");
-                    user_info.subscription.current_state.payment_currency =
-                        get_string_field(foreign_state, "payment_currency");
-                    user_info.subscription.current_state.payment_provider =
-                        get_string_field(foreign_state, "payment_provider");
-                    user_info.subscription.current_state.payment_status =
+                    tracker_info
+                        .user
+                        .subscription
+                        .current_state
+                        .payment_currency = get_string_field(foreign_state, "payment_currency");
+                    tracker_info
+                        .user
+                        .subscription
+                        .current_state
+                        .payment_provider = get_string_field(foreign_state, "payment_provider");
+                    tracker_info.user.subscription.current_state.payment_status =
                         get_string_field(foreign_state, "payment_status");
-                    user_info.subscription.current_state.plan_id =
+                    tracker_info.user.subscription.current_state.plan_id =
                         get_i32_field(foreign_state, "plan_id");
-                    user_info.subscription.current_state.plan_type =
+                    tracker_info.user.subscription.current_state.plan_type =
                         get_string_field(foreign_state, "plan_type");
-                    user_info.subscription.current_state.subscription_status =
+                    tracker_info
+                        .user
+                        .subscription
+                        .current_state
+                        .subscription_status =
                         get_string_field(foreign_state, "subscription_status");
                 }
             }
         }
-    }
 
-    (device_info, user_info)
+        if let Some(foreign_app_info) = foreign_context.get("application") {
+            tracker_info.application.name = get_string_field(foreign_app_info, "name");
+            tracker_info.application.version = get_string_field(foreign_app_info, "version");
+            tracker_info.application.platform = get_string_field(foreign_app_info, "platform");
+            tracker_info.application.arch = get_string_field(foreign_app_info, "arch");
+
+            if let Some(foreign_app_config_info) = foreign_app_info.get("config") {
+                if let Some(foreign_app_config_current_state_info) =
+                    foreign_app_config_info.get("current_state")
+                {
```

Comment:
```suggestion
            if let (Some(foreign_app_config_info), Some(foreign_app_config_current_state_info)) = (foreign_app_info.get("config", foreign_app_config_info.get("current_state")) {
```
It would be one nesting less

Reply:
not sure about the readability gains.. it's dependant on the order of execution (`foreign_app_config_info` doesn't exist)

Reply:
I mean you can skip this, however, you actually need both of them, regardless of the order